docs/modules/ROOT/pages/configure-cmos.adoc

```diff
@@ -0,0 +1,230 @@
+
+[NOTE]
+To pick up any configuration changes after the Prometheus service has started, you may need to reload the config files using the `reload` endpoint.
+
+Below are the environment variables used for Prometheus configuration.
+To read more about Prometheus configuration, see the https://prometheus.io/docs/prometheus/latest/configuration/configuration/[official documentation website^].
+
+[console]
+----
+# Prometheus configuration file
+PROMETHEUS_CONFIG_FILE=${PROMETHEUS_CONFIG_FILE:-/etc/prometheus/prometheus-runtime.yml}
+
+# Prometheus configuration file template
+PROMETHEUS_CONFIG_TEMPLATE_FILE=${PROMETHEUS_CONFIG_TEMPLATE_FILE:-/etc/prometheus/prometheus-template.yml}
+
+# Prometheus url sub path
+PROMETHEUS_URL_SUBPATH=${PROMETHEUS_URL_SUBPATH-/prometheus/}
+
+# Prometheus storage path
+PROMETHEUS_STORAGE_PATH=${PROMETHEUS_STORAGE_PATH-/prometheus}
+
+# Microlith configuration
+PROMETHEUS_DYNAMIC_INTERNAL_DIR=${PROMETHEUS_DYNAMIC_INTERNAL_DIR:-/etc/prometheus/couchbase/monitoring/}
+----
+
+=== Endpoints
+
+In CMOS, a https://prometheus.io/docs/prometheus/latest/configuration/configuration/#file_sd_config[file-based service discovery^] approach is used to support adding/removing endpoints.
+These endpoints are used to dynamically scrape metrics.
+
+Based on which services are enabled, the microlith will construct a set of dynamic endpoints.
+These endpoints are created for each service by adding a JSON file to the `${PROMETHEUS_DYNAMIC_INTERNAL_DIR:-/etc/prometheus/couchbase/monitoring/}`.
+
+To add Couchbase Server endpoints, create a similar JSON format file like the example in the `/etc/prometheus/couchbase/custom/` directory mounted on the container.
+In order to always stay current with the system state, this file is periodically rescanned to add or remove targets.
+
+While this enables the addition of a new scrape target, it will not automatically add dashboards to Grafana. The dashboards will need to be added manually for the relevant scrape target.
+
+[console]
+----
+[
+    {
+      "targets": [
+        "exporter:9091"
+      ],
+      "labels": {
+        "job": "db1",
+        "container": "exporter"
+      }
+    }
+]
+----
+
+You can set the authentication credentials for your Couchbase Server clusters using the `$CB_SERVER_AUTH_USER` and `$CB_SERVER_AUTH_PASSWORD` environment variables.
+Note that currently we do not support using different credentials for multiple clusters.
+
+[NOTE]
+This is a legacy way of configuring Prometheus targets and we recommend not to use this approach.
+You can use CMOS UI to perform the same using the "Add Cluster" option present on the landing page.
+
+.Add cluster image
+image::add-cluster-vm.png[]
+
+== Loki
+
+Configuration file for the log aggregation system Loki is present in the YAML file located at  `/etc/loki/local-config.yaml` inside the container.
```

Comment:
```suggestion
Configuration file for the log aggregation system Loki is present in the YAML file located at  `/etc/loki/config.yaml` inside the container.
```

Slightly outdated - this is the correct path (the console snippet below is correct).